src/witan/gateway/components/query_router.clj

```diff
@@ -1,32 +1,45 @@
 (ns witan.gateway.components.query-router
   (:require [com.stuartsierra.component :as component]
             [taoensso.timbre            :as log]
-            [ring.util.http-response    :refer [ok not-found failed-dependency]]
             [witan.gateway.protocols    :as p :refer [RouteQuery]]
-            [clj-http.client            :as client]))
-
+            [clj-http.client            :as client]
+            [clojure.data.codec.base64  :as b64]))
 
+(defn detect-namespace
+  [p]
+  (let [x (-> p first first)]
+    (if (keyword? x)
+      (keyword (namespace x))
+      (when (and (coll? x)
+                 (keyword? (first x)))
+        (keyword (namespace (first x)))))))
```

Comment:
It returns `nil` and fails the `contains?` check (here)[https://github.com/MastodonC/witan.gateway/blob/0db897d979cbe561205d62d42300bae54f649bb8/src/witan/gateway/components/query_router.clj#L22]